apm-agent-plugins/apm-servlet-plugin/src/main/java/co/elastic/apm/agent/servlet/ServletTransactionHelper.java

```diff
@@ -144,13 +144,38 @@ public static void setUsernameIfUnset(@Nullable String userName, TransactionCont
         }
     }
 
+    public String normalizeServletPath(String requestURI, @Nullable String contextPath, @Nullable String servletPath, @Nullable String pathInfo) {
+        String path = servletPath;
+        if (path == null || path.isEmpty()) {
```

Comment:
We'd expect this case to be relatively rare, right? It does cause some allocations but if it's an edge case I think that's fine.

Reply:
Yes, this is definitely an edge case as most of the time we have `servletPath` provided as we usually expect.